test/tools/fixtures/process_resource_script.in.js

```diff
@@ -78,12 +78,14 @@ async function main() {
 main()
   .then(() => {})
   .catch(e => {
-    log({ exitCode: 1, error: util.inspect(e) });
+    log({ exitCode: 1, error: { message: e.message, stack: e.stack } });
   });
 
 setTimeout(() => {
   // this means something was in the event loop such that it hung for more than 10 seconds
   // so we kill the process
+  log({ newLibuvResources: getNewLibuvResourceArray() });
+  log({ exitCode: 99, error: { message: 'Process timed out: resources remain in the event loop.' } });
```

Comment:
Because there's no stack prop, we attach "undefined" to the stack that gets thrown in the parent. I think we want to remove this log line and instead special case 99 in the parent so that we can log the resources you log above here so we can see what possible resources are still holding open the process.

Reply:
Yep, I've stringified the resources and printed them with both non-zero exit codes.